src/packages/connections/sockets/logic/creators/createSocketCommunicator.ts

```diff
@@ -113,6 +120,51 @@ export default function createSocketCommunicator({
     return send(data, processes, abort);
   }
 
+  async function scatter(
+    data: unknown[],
+    sendCount: number,
+    receiveBuf: unknown[],
+    receiveCount: number,
```

Comment:
question(blocking): why do we need both `receiveCount` and `sendCount`? tbh, i don't see a case where this numbers can differ